plugin/storage/es/options_test.go

```diff
@@ -148,3 +152,29 @@ func TestMaxDocCount(t *testing.T) {
 		})
 	}
 }
+
+func TestIndexDateSeparator(t *testing.T) {
+	testCases := []struct {
+		name           string
+		flags          []string
+		wantDateLayout string
+	}{
+		{"not defined (default)", []string{}, "2006-01-02"},
+		{"empty separator", []string{"--es.index-date-separator="}, "20060102"},
+		{"dot separator", []string{"--es.index-date-separator=."}, "2006.01.02"},
+		{"crossbar separator", []string{"--es.index-date-separator=-"}, "2006-01-02"},
+		{"backslash separator", []string{"--es.index-date-separator=/"}, "2006/01/02"},
+		{"multiple characters separator", []string{"--es.index-date-separator=' '"}, "2006' '01' '02"},
```

Comment:
I actually meant empty string `''` to make it clear that empty quoted string is not treated as an empty separator :)

Reply:
Updated. Please check if the description of the test is accurate.